napari/conftest.py

```diff
@@ -470,3 +472,166 @@ def _mock_app():
             yield app
         finally:
             Application.destroy('test_app')
+
+
+def _get_calling_place(depth=1):
+    if hasattr(sys, "_getframe"):
+        frame = sys._getframe(1 + depth)
+        return f"{frame.f_code.co_filename}:{frame.f_code.co_firstlineno}"
+    return ""
+
+
+@pytest.fixture
+def dangling_threads(monkeypatch):
+    from qtpy.QtCore import QThread
+
+    base_start = QThread.start
+    thread_dkt = WeakKeyDictionary()
+
+    def my_start(self, priority=QThread.InheritPriority):
+        thread_dkt[self] = _get_calling_place()
+        base_start(self, priority)
+
+    monkeypatch.setattr(QThread, 'start', my_start)
+    yield
+
+    dangling_threads = []
+
+    for thread, calling in thread_dkt.items():
+        try:
+            if thread.isRunning():
+                dangling_threads.append((thread, calling))
+        except RuntimeError as e:
+            if "wrapped C/C++ object of type QThread" not in e.args[0]:
+                raise
+
+    for thread, _ in dangling_threads:
+        thread.quit()
+        thread.wait(2000)
+
+    assert not dangling_threads, "thread called in:\n" + "\n".join(
```

Comment:
Consider adding a descriptive adjective to the assertion message (e.g. `unfinished thread ...` or `dangling thread ...`). Same comment for other similar assertion messages in this PR.

Reply:
Could you check the current one?

Reply:
Love the new messages! Just made a couple of typo suggestions.

Reply:
Merged. Thanks